test/uk/gov/hmrc/incometaxsubscriptionselfemployedfrontend/controllers/BusinessNameControllerSpec.scala

```diff
@@ -0,0 +1,114 @@
+ * limitations under the License.
+ */
+
+package uk.gov.hmrc.incometaxsubscriptionselfemployedfrontend.controllers
+
+import play.api.mvc.{Action, AnyContent}
+import play.api.test.FakeRequest
+import play.api.test.Helpers._
+import uk.gov.hmrc.http.InternalServerException
+import uk.gov.hmrc.incometaxsubscriptionselfemployedfrontend.connectors.httpparser.GetSelfEmploymentsHttpParser.UnexpectedStatusFailure
+import uk.gov.hmrc.incometaxsubscriptionselfemployedfrontend.connectors.httpparser.PostSelfEmploymentsHttpParser
+import uk.gov.hmrc.incometaxsubscriptionselfemployedfrontend.connectors.httpparser.PostSelfEmploymentsHttpParser._
+import uk.gov.hmrc.incometaxsubscriptionselfemployedfrontend.connectors.mocks.MockIncomeTaxSubscriptionConnector
+import uk.gov.hmrc.incometaxsubscriptionselfemployedfrontend.forms.{BusinessNameForm, BusinessStartDateForm}
+import uk.gov.hmrc.incometaxsubscriptionselfemployedfrontend.models.{BusinessNameModel, BusinessStartDate}
+import uk.gov.hmrc.incometaxsubscriptionselfemployedfrontend.utilities.TestModels._
+
+import scala.concurrent.Future
+
+class BusinessNameControllerSpec extends ControllerBaseSpec
+  with MockIncomeTaxSubscriptionConnector {
+  override val controllerName: String = "BusinessNameController"
+  override val authorisedRoutes: Map[String, Action[AnyContent]] = Map(
+    "show" -> TestBusinessNameController$.show(),
+    "submit" -> TestBusinessNameController$.submit()
+  )
+
+  object TestBusinessNameController$ extends BusinessNameController(
+    mockMessagesControllerComponents,
+    mockIncomeTaxSubscriptionConnector,
+    mockAuthService,
+    mockLanguageUtils
+  )
+
+  def modelToFormData(businessNameModel: BusinessNameModel): Seq[(String, String)] = {
+    BusinessNameForm.businessNameValidationForm.fill(businessNameModel).data.toSeq
+  }
+
+  "show" should {
+    "return ok (200)" when {
+      "the connector returns data" in {
+        mockAuthSuccess()
+        mockBusinessGetSelfEmployments("businessName")(Future.successful(Right(Some(BusinessNameModel("Business")))))
+        val result = TestBusinessNameController$.show()(FakeRequest())
+        status(result) mustBe OK
+        contentType(result) mustBe Some("text/html")
+      }
+      "the connector returns no data" in {
+        mockAuthSuccess()
+        mockBusinessGetSelfEmployments("businessName")(Future.successful(Right(None)))
+        val result = TestBusinessNameController$.show()(FakeRequest())
+        status(result) mustBe OK
+        contentType(result) mustBe Some("text/html")
+      }
+      "Throw an internal exception error" when {
+        "the connector returns an error" in {
+          mockAuthSuccess()
+          mockBusinessGetSelfEmployments("businessName")(Future.successful(Left(UnexpectedStatusFailure(INTERNAL_SERVER_ERROR))))
+          intercept[InternalServerException](await(TestBusinessNameController$.show()(FakeRequest())))
+        }
+      }
+
+    }
+
+    authorisationTests()
```

Comment:
You only need one authorisationTests() and it should be called outside of tests, see other controller spec